packages/client/src/modules/chat/containers/Chat.native.jsx

```diff
@@ -26,6 +42,43 @@ class Chat extends React.Component {
     message: ''
   };
 
+  componentWillReceiveProps() {
+    // Check if props have changed and, if necessary, stop the subscription
+    if (this.subscription) {
+      this.subscription();
+      this.subscription = null;
+    } else {
+      // Subscribe or re-subscribe
+      this.subscribeToMessages();
+    }
+  }
+
+  subscribeToMessages = () => {
+    const { subscribeToMore } = this.props;
+
+    this.subscription = subscribeToMore({
+      document: MESSAGES_SUBSCRIPTION,
+
+      updateQuery: (
+        prev,
+        {
+          subscriptionData: {
+            data: {
+              messagesUpdated: { mutation, node }
+            }
+          }
+        }
+      ) => {
+        let newResult = prev;
```

Comment:
Since we stick to a functional paradigm, always try to avoid mutable variables. It is just a matter of a good style.
I know that this is the solution from other modules, but it doesn't mean it is OK.
Here how we can do it without redundant variables:
`
if (mutation === 'CREATED') {
   return AddMessage(prev, node);
}

return prev;
`

Reply:
@evgenijolshanskij This introduces other problem - returns in arbitrary points of the function which makes code readability worse and is much more error prone than mutable variables because of that. If to rewrite this function to get rid of mutable variables it should be rather done like this:
```
) => (mutation === 'CREATED' ? AddMessage(prev, node) : prev)
```
or at least like this:
```
) => {
  return mutation === 'CREATED' ? AddMessage(prev, node) : prev;
}
```
Generally function should have as few returns as possible and they should be either in the begging of the function (when checking parameters and returning early, though this should be reasonably avoided too) or as the last statement of the function (which is ideal), never in the middle. This is of a higher priority then mutable variables considerations. Mutable variable consideration should be take care too, but without sacrificing returns policies.

Reply:
@evgenijolshanskij As a general rule we can also stop adding new feature when we see bad code pattern somewhere in the kit and instead switch to hunting for this pattern throughout all the kit code and fixing it, so that our contributors weren't confused about what patterns in the code they should use.

Reply:
@vlasenko Everything you said makes complete sense. In this specific code snippet, unfortunately, we cannot use this approach ```) => (mutation === 'CREATED' ? AddMessage(prev, node) : prev)```, because we have at least 3 mutations types.
Probably, for the clarity we could use ```switch``` constuction with returns in every case.